### PR TITLE
Closed loop 2083885

### PR DESCRIPTION
**Pull request by pranavprakash20**
# Description
Automate https://issues.redhat.com/browse/RHCEPH-4275 
Polarion id : CEPH-83575014
Test result link : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-XTZS7F/

Updated Test result:

TEST NAME                        TEST DESCRIPTION                                               DURATION                         STATUS                    COMMENTS
setup install pre-requisistes    Setup phase to deploy the required pre-requisites for runnin   0:06:22.067207                   Pass                             
Cluster deployment               Cluster deployment with latest released version of 5.x         0:20:31.339054                   Pass                             
Configure client                 Configure client with admin keyring.                           0:00:20.156376                   Pass                             
Upgrade along with IOs           Running upgrade and i/o's parallelly                           0:11:31.437960                   Pass                             
test non default rgw pool crea   Verifies if default rgw pool is created when RGW service is    0:00:05.406168                   Pass                             

Test result link : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-MUALG2/

- [x] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [x] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [x] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test

### CODE REVIEW

File: tests/cephadm/test_validate_osd_pool.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+CLUSTER_STATE = [
+    "ceph osd df",
+    "ceph osd tree",
+]
+
+
+def run(ceph_cluster, **kw):
+    """
+    Ceph-admin module to manage ceph-osd service
```
Comment (psathyan) [inline tests/cephadm/test_validate_osd_pool.py:17]: Not sure if the docString is correct as the test module name is validate osd pool. Here the description is about managing the daemon.

File: tests/cephadm/test_validate_osd_pool.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    orch = OSD(cluster=ceph_cluster, **config)
+
+    try:
+        method = fetch_method(orch, command)
+        log.info("output og fetch_method : %s" %method)
+        method(config)
+    finally:
+        # Get cluster state
+        get_cluster_state(orch)
+    return 0
```
Comment (psathyan) [inline tests/cephadm/test_validate_osd_pool.py:39]: Please end with a newline character.

File: tests/cephadm/test_validate_osd_pool.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        ceph_cluster (ceph.ceph.Ceph): Ceph cluster object
+        kw: test data
+
+    check ceph.ceph_admin.osd for test config
+    """
+    config = kw.get("config")
+
+    # Manage Ceph using ceph-admin orchestration
+    command = config.pop("command")
+    log.info("Executing service %s service" % command)
```
Comment (psathyan) [inline tests/cephadm/test_validate_osd_pool.py:29]: The command passed is `ceph osd pool ls detail`... which is not a service. Could we fix the log message.

**Review by psathyan — COMMENTED** [review]
Please attach logs of a unit test run.

**Comment by psathyan** [comment]
Also please squash all your commits into one.

File: tests/cephadm/test_validate_osd_pool.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+from ceph.ceph_admin.common import fetch_method
```
Comment (sunilkumarn417) [inline tests/cephadm/test_validate_osd_pool.py:1]: I think we can use `test_cephadm.py` for command execution, if it is cephadm shell command.

File: tests/cephadm/test_validate_osd_pool.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    """
+    config = kw.get("config")
+
+    # Manage Ceph using ceph-admin orchestration
+    command = config.pop("command")
+    log.info("Executing service %s service" % command)
+    orch = OSD(cluster=ceph_cluster, **config)
+
+    try:
+        method = fetch_method(orch, command)
```
Comment (sunilkumarn417) [inline tests/cephadm/test_validate_osd_pool.py:33]: IMO, `command: ceph osd pool ls detail` may fail to fetch command value method.

File: suites/pacific/upgrades/tier-1_upgrade_test-4x-to-5x-container.yaml [hunk 6]
```diff
@@ -157,3 +157,13 @@ tests:
...
     desc: Perform rgw tests
+
+- test:
+    name: validate-rgw-pool
+    module: test_validate_osd_pool.py
+    polarion-id: CEPH-83575014
+    config:
+      cmd: ceph osd pool ls detail
+      sudo: True
+    desc: Validate osd pool after upgrade
```
Comment (psathyan) [inline suites/pacific/upgrades/tier-1_upgrade_test-4x-to-5x-container.yaml:168]: The test is specific to verifying if `.rgw.root` data pool... I believe we can accomplish this by using the following lines
```
- test:
    name: test non default rgw pool creation
    module: exec.py
    polarion-id: 
    config:
      check_status: true
      cephadm: true
      commands:
        - "ceph osd pool ls detail | grep -v .rgw.root"
    desc: Verifies if default rgw pool is created when RGW service is deployed.
```

File: tests/cephadm/test_validate_osd_pool.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
+# Module to assist in executing a sequence of command.
```
Comment (psathyan) [inline tests/cephadm/test_validate_osd_pool.py:1]: IMO, we don't need to create a separate module for this validation unless you plan to build addition tasks around this test case.

If you still feel a need for the module... could we rename it to `test_validate_default_rgw_pool.py`

**Comment by psathyan** [comment]
@pranavprakash20 there is a conflict, please resolve the conflict and let's working towards closing this PR.

File: suites/pacific/upgrades/tier-1_upgrade_cephadm_without_rgw.yaml [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+                type: rados
+                pool_per_client: true
+                pg_num: 128
+                duration: 10
+              verify_cluster_health: true
+            destroy-cluster: false
+      desc: Running upgrade and i/o's parallelly
+      abort-on-fail: true
+
+  - test:
```
Comment (pranavprakash20) [inline suites/pacific/upgrades/tier-1_upgrade_cephadm_without_rgw.yaml:195]: The test requires a ceph cluster with a lower version of ceph 5 and no rgw pool existing. Adding the suggested changes to a new suite without rgw configured. A corresponding conf is also created without the rgw details.

**Comment by pranavprakash20** [comment]
> @pranavprakash20 there is a conflict, please resolve the conflict and let's working towards closing this PR.

@psathyan The conflict has been resolved. The tc has been revisited (the test steps has been modified as it serves better when tested on a lower version of ceph5 without rgw pool and then upgrade to latest)

**Review by vdas-redhat — APPROVED** [review]
LGTM

**Review by psathyan — APPROVED** [review]

**Pull request merged**